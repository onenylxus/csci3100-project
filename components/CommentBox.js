// Import
import React from 'react';
import { Image, Text, View } from 'react-native';
import Style from '../assets/style';

// Export Comment Box
export default function CommentBox({ comment }) {
  return (
    <View style={Style.commentBox}>
      <View style={{ flexDirection: 'row' }}>
        <Image
          style={{
            width: 48,
            height: 48,
            margin: 10,
            borderRadius: 28,
          }}
          source={require('../assets/images/profile.png')}
        />
        <View style={{ flexDirection: 'column', marginTop: 15 }}>
          <Text>{comment.username}</Text>
          <Text>{comment.content}</Text>
        </View>
      </View>
    </View>
  );
}
