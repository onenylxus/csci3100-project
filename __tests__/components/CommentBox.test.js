// Import
import React from 'react';
import { render } from '@testing-library/react-native';

// Import target component
import CommentBox from '../../components/CommentBox';

// Mock FontAwesome icons
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));

// Run jest tests
describe('CommentBox', () => {
  let element;

  beforeEach(() => {
    // Render
    element = render(<CommentBox />);
  });

  it('matches snapshot', () => {
    expect(element.toJSON()).toMatchSnapshot();
  });
});
