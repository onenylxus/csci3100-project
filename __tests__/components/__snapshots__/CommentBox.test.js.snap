// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentBox matches snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#f0f0f0",
      "borderRadius": 10,
      "borderWidth": 1,
      "flexDirection": "column",
      "padding": 4,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "marginTop": 5,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "borderRadius": 28,
              "height": 32,
              "marginHorizontal": 8,
              "marginTop": 4,
              "width": 32,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            testac
          </Text>
          <Text
            style={
              Object {
                "fontSize": 10,
              }
            }
          >
            00:00  01/05/2021
          </Text>
          <Text>
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
